syz-verifier/stats.go

```diff
@@ -92,21 +162,22 @@ func (stats *Stats) getCallStatsTextDescription(call string) string {
 		"\t↳ mismatches of %s / total number of mismatches: "+
 		"%d / %d (%0.2f %%)\n"+
 		"\t↳ %d distinct states identified: %v\n", syscallName, syscallName, syscallName, mismatches, occurrences,
-		getPercentage(mismatches, occurrences), syscallName, mismatches, stats.TotalCallMismatches,
-		getPercentage(mismatches, stats.TotalCallMismatches), len(syscallStat.States), stats.getOrderedStates(syscallName))
+		getPercentage(mismatches, occurrences), syscallName, mismatches, stats.TotalCallMismatches.Get(),
+		getPercentage(mismatches, stats.TotalCallMismatches.Get()),
+		len(syscallStat.States), stats.getOrderedStates(syscallName))
 }
 
-func (stats *Stats) totalCallsExecuted() int64 {
-	var t int64
-	for _, cs := range stats.Calls {
+func (stats *Stats) totalCallsExecuted() uint64 {
+	var t uint64
+	for _, cs := range stats.calls {
```

Comment:
Here and in other places - I think there's still a risk of writing to the map and reading from it at the same time. Golang's implementation does not support it.

Maybe create some custom data structure that would wrap the list of calls and do the necessary locking inside? Then you won't have to care about mutexes while accessing `Stats`.

Reply:
Done. PTAL.